packages/core/src/workflow/context.ts

```diff
@@ -62,50 +122,140 @@ export class Context {
     return null;
   }
 
-  sendEvent(message: WorkflowEvent, step?: StepFunction): void {
-    const stepName = step?.name ? `step ${step.name}` : "all steps";
+  #pushEvent(event: WorkflowEvent, step: StepFunction<Start>): void {
+    const stepName = step.name ? `step ${step.name}` : "all steps";
     if (this.#verbose) {
-      console.log(`Sending event ${message} to ${stepName}`);
+      console.log(`Sending event ${event} to ${stepName}`);
     }
-    if (step === undefined) {
-      for (const queue of this.#queues.values()) {
-        queue.push(message);
-      }
-    } else {
-      if (!this.#workflow.hasStep(step)) {
-        throw new Error(`Step ${step} does not exist`);
-      }
 
-      if (!this.#queues.has(step)) {
-        this.#queues.set(step, []);
-      }
-      this.#queues.get(step)!.push(message);
+    if (!this.#steps.has(step)) {
+      throw new Error(`Step ${step} does not exist`);
     }
+
+    this.#queue.push(event);
   }
 
-  getNextEvent(step: StepFunction): WorkflowEvent | undefined {
-    const queue = this.#queues.get(step);
-    if (queue && queue.length > 0) {
-      return queue.shift();
+  // make sure it will only be called once
+  #iterator: AsyncGenerator<WorkflowEvent, void, void> | null = null;
+  #signal: AbortSignal | null = null;
+  get #iteratorSingleton(): AsyncGenerator<WorkflowEvent, void, void> {
+    if (this.#iterator === null) {
+      this.#iterator = this.#createStreamEvents();
     }
-    return undefined;
+    return this.#iterator;
   }
 
-  writeEventToStream(event: WorkflowEvent): void {
```

Comment:
This seems different:
The original code only sends explicitly sent events (via writeEventToStream) to streamEvents

Reply:
that's why there's the streamingQueue

Reply:
remove this now. you can customize your context data to have such queue.